pkg/scheduler/queue/multi_queuing_algorithm_tree_queue.go

```diff
@@ -126,37 +129,56 @@ type Node struct {
 	childrenChecked  int
 }
 
-func newNode(name string, depth int, da QueuingAlgorithm) (*Node, error) {
+func newNode(name string, depth int, da QueuingAlgorithm, isLeaf bool) (*Node, error) {
```

Comment:
could we pass in height instead of depth, and have `height == 0` be the check for whether it is a leaf?